proxy.py

```diff
@@ -11,61 +11,96 @@ def __init__(self):
         super().__init__()
         self.logger = get_nio_logger("ConfigProxy")
 
-    def get_version(self, url_prefix, config_id, apikey):
-        """
-        Retrieves the latest instance configuration version id
+    def get_instance_config_ids(self, url_prefix, instance_id, apikey):
+        """ Gets the conf id and conf version id the instance should be running
+
+        Args:
+            url_prefix (str): url prefix to product api
+            instance_id (str): instance id
+            apikey (str): api key
 
         Returns: an object with format
             {
+                "instance_configuration_id": "uuid..",
                 "instance_configuration_version_id": "uuid..",
                 "version_num": "v1.0.0",
                 "created_at": "2018-03-14 12:12:12"
             }
         """
+        url = "{}/instances/{}/configuration".format(url_prefix,
+                                                     instance_id)
+        return self._request(
+            requests.get, url, apikey, "Failed to get configuration version "
+                                       "the instance should be running")
 
-        try:
-            url = "{}/{}/versions/latest".format(url_prefix, config_id)
-            response = requests.get(
-                url,
-                headers=self._get_headers(apikey)
-            )
-            if response.ok:
-                return response.json()
-        except requests.exceptions.ConnectionError:
-            self.logger.exception("Failed to get configuration version")
+    def notify_instance_config_ids(self, url_prefix, instance_id, config_id,
+                                   config_version_id, apikey):
+        """ Posts instance config ids
 
-    def load_configuration(self, url_prefix, \
-                           config_id, config_version_id, apikey):
-        """ 
-        Retrieves an instance configruation by a instance_configuration_id 
-        and a instance_configuration_version_id
+        Args:
+            url_prefix (str): url prefix to product api
+            instance_id (str): instance id
+            config_id (str): instance configuration id
+            config_version_id (str): instance configuration version id
+            apikey (str): api key
+        """
+
+        url = "{}/instances/{}/configuration".format(url_prefix,
+                                                     instance_id)
+        return self._request(
+            requests.post, url, apikey, "Failed to post new configuration "
+                                        "version instance is running",
+            json={
+                "reported_configuration_id": config_id,
+                "reported_configuration_version_id": config_version_id
```

Comment:
Per API-444, this endpoint will also accept `deployment_id`, `status`, and `message` to update the product api when a deployment to an instance has succeeded or failed along with a message `updated [] services` or `failed to install [] block`